gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -271,35 +270,38 @@ public Completable deleteOfflineArea(String offlineAreaId) {
         .map(ImmutableList::asList)
         .flatMap(Flowable::fromIterable)
         .firstOrError()
-        .map(this::identifyAndHandleSource)
+        .flatMap(this::identifyAndHandleSource)
         .doOnError(t -> Timber.e(t, "Couldn't retrieve basemap sources for the active project"));
   }
 
-  private ImmutableList<TileSet> identifyAndHandleSource(BaseMap baseMap) throws IOException {
+  private Single<ImmutableList<TileSet>> identifyAndHandleSource(BaseMap baseMap)
```

Comment:
It's not immediately clear what this method does from its name. My understanding is that it gets the TileSets comprising the base map (and, that identifying the type of BaseMap is an implementation detail included in that process). Perhaps something like `getTilesSets(BaseMap)` would make more sense? JavaDoc should mention that this may make a remote call.

Reply:
Good call, thanks! Done!